File: 그리디/전승훈/pro_단속카메라.swift
import Foundation

func solution(routes: [[Int]]) -> Int {
    let sortedRoutes = routes.sorted { $0[1] < $1[1] }
    print(sortedRoutes)
    var camera = -99999
    var count = 0
    
    sortedRoutes.forEach { route in
        if route[0] > camera {
            print("route[0]: \(route[0]) vs camera: \(camera)")
            camera = route[1]
            count += 1
        }
    }
    return count
}

File: 그리디/김상효/pro_단속카메라.py
def solution(routes):
    answer = 1
    sortedRoutes = sorted(routes, key = lambda x: -x[0])
    checkPoint = sortedRoutes[0][0]
    
    checked = [0 for _ in range(len(routes))]

    for i in range(len(sortedRoutes)):
        
        if sortedRoutes[i][1] < checkPoint:
            answer += 1
            checkPoint = sortedRoutes[i][0]

    return answer

routes = [[-20,-15], [-14,-5], [-18,-13], [-5,-3]]
print(solution(routes))